### DISCUSSION TITLE
Backpack in a subdomain

### DISCUSSION THREAD

**Discussion by cdo9** [Laravel-Backpack/community-forum#887] [Q&A (Help)] [answered]
I am currently creating a "multi-site", this is the routing I want to achieve :

```
mydomain1.com - The Website n°1
mydomain2.com - The Website n°2
admin.mydomain3.com - The Backpack admin
```

My route file look like this : 
```
Route::domain('mydomain1.com')->group(function () {
// Website n°1 routes
}
Route::domain('mydomain2.com')->group(function () {
// Website n°2 routes
}
Route::group([
        'prefix' => config('backpack.base.route_prefix', 'backpack'),
        'middleware' => array_merge(
            (array) config('backpack.base.web_middleware', 'web'),
            (array) config('backpack.base.middleware_key', 'admin')
        ),
        'namespace' => 'App\Http\Controllers\Admin',
        'domain' => 'admin.mydomain3.com',
    ], function () { // custom admin routes
// Backpack routes
});
```

1. If I remove prefix, backpack routes are also available in my other domains, but CRUDs are working well.
2. If I set a prefix, backpack routes are still available in my other domains, but CRUDs are not working (404 errors)

Does Backpack really need the APP_URL ? I tried to set it to "empty" and to "http://admin.mydomain3.com" with no luck. In both case no difference between 1 and 2.

**Option 1** is my favorite, and CRUDS are OK on it, but why the routes are also available in my other domains ?

### Why I want to achieve this ?
For a website I am building, 1 admin for 2(+) eCommerce, some products are shared between the 2(+) sites, some are not.

**Comment by jcastroa87** [comment]
Hello @cdo9 

First, i will suggest that you better separate front-office (for ecommerce customers) from back-office (for admins).

As I understand, with your config, you still have /dashboard, /login, etc, accessible from other domains, that is because Backpack automatically creates these routes, you need to disable it from the config, check in our [documentation](https://backpackforlaravel.com/docs/6.x/base-how-to#customizing-the-auth-controllers) and after this need to create manually these routes.

Let me know if you need any other help.

Cheers.

**Reply by cdo9** [reply]
Hello @jcastroa87,

Thanks for your help. I didn't think about disabling auth & dashboard routes. This is working now.

I don't want to separate front-office from back-office since they share too much code but thanks for your suggestion :)

**Comment by jcastroa87** [comment]
I'm glad you solve the problem. I will close the issue, but please feel free to re-open it or create a new one.

Cheers.

**Comment by pxpm** [comment, marked as answer]
Hello @cdo9 

Backpack register some extra routes not exposed in `custom.php`, they are in `base.php`: https://github.com/Laravel-Backpack/CRUD/blob/main/src/routes/backpack/base.php

To customize them, you can copy the file from the vendor folder into your `routes/backpack/base.php` and do the needed changes there:
```diff
Route::group(
    [
        'namespace' => 'Backpack\CRUD\app\Http\Controllers',
        'middleware' => config('backpack.base.web_middleware', 'web'),
        'prefix' => config('backpack.base.route_prefix'),
+       'domain' => 'admin.mydomain3.com
    ],
    function () {
      // ... the routes setup
    });
```

That way those routes will only be available under that domain. 

Hope that helps.

Cheers

**Reply by cdo9** [reply]
Hello @pxpm and thanks! yes it's better that way so I can keep the login route for my admins :)
